backend/src/map_model.rs

```diff
@@ -994,8 +994,11 @@ impl MapModel {
             .map(|r| r.main_road_penalty != main_road_penalty)
             .unwrap_or(true)
         {
-            let router_before_with_penalty =
-                Router::new(&self.router_input_before(), main_road_penalty);
+            let router_before_with_penalty = if self.router_before.main_road_penalty == main_road_penalty {
+                self.router_before.clone()
```

Comment:
This shaves off ~750ms from the 3.000 s initial load time if you haven't already loaded the router.

e.g. refresh the page, click "predict impact".